joomla/joomla-cms
# [4.2] Allow to disable session metadata tracking for guest users

### Summary of Changes

This should improve performance for sites that does not use User registration/login feature.

We already have an option to disable "Track Session Metadata",
However it more usefull to be able to disable it only for non registered users, while keep track of metadata for loged in users.
That what I made here.

### Testing Instructions

Enable debug with DB queries enabled. 
Visit the site frontend (do not login) and check how much DB queries.

Apply patch.
Go to global configuration and set "Track Session Metadata for non registered Users" to "No".
Visit the site frontend again (do not login) and check how much DB queries.

### Expected result AFTER applying this Pull Request
All should continue work as before, without any errors.

And it should reduce 2db query.


### Documentation Changes Required

Yes, I guess, need a description for this feature.
